### PR TITLE
Enable dts-trimming for packages/drivers

### PR DESCRIPTION
**Pull request by jikim-msft**
#### Description

Related: [18176](https://github.com/microsoft/FluidFramework/pull/18176/files)

This PR adds type-trimming for `packages/drivers` directory. The changes introduced are: 
- Add npm `copyfiles` package as `devDependency`
- Add `dtsRollup` in `api-extractor.json`
- Enable both type-trimming for both `ESM` & `CJS`
- Make `fluidBuild` to fetch dependencies from the scripts included in `dependsOn`

### CODE REVIEW

**Review by tylerbutler — APPROVED** [review]

File: packages/drivers/fluidapp-odsp-urlResolver/package.json
Comment (alexvy86) [inline packages/drivers/fluidapp-odsp-urlResolver/package.json:1]: Need to add build:docs and ci:build:docs scripts.

File: packages/drivers/odsp-urlResolver/package.json
Comment (alexvy86) [inline packages/drivers/odsp-urlResolver/package.json:1]: Also build:docs and ci:build:docs scripts.

File: packages/drivers/odsp-urlResolver/src/urlResolver.ts [hunk 1]
```diff
@@ -36,6 +39,10 @@ export class OdspUrlResolver implements IUrlResolver {
 	}
 }
 
+// eslint-disable-next-line jsdoc/require-description
```
Comment (alexvy86) [inline packages/drivers/odsp-urlResolver/src/urlResolver.ts:42]: Does it complain without this line? Seems like the class above didn't and it also doesn't have description.

File: packages/drivers/routerlicious-urlResolver/package.json
Comment (alexvy86) [inline packages/drivers/routerlicious-urlResolver/package.json:1]: `build:docs` and `ci:build:docs`

File: packages/drivers/odsp-urlResolver/src/urlResolver.ts (hunk 1 above)
Comment (jikim-msft) [reply packages/drivers/odsp-urlResolver/src/urlResolver.ts:42]: image

Linter is weird sometimes 😞. Doesn't complain about the missing description for the class but does so for the function. The function was pretty straightforward, so I added the description instead.

**Pull request merged**